### PR TITLE
Feature/NR-36 add trending games

### PR DESCRIPTION
**Pull request by lukatarman**
Related issue: #3

### CODE REVIEW

File: src/infrastructure/repositories/games.repository.js [hunk 1]
```diff
@@ -134,4 +134,176 @@ export class GamesRepository {
       ])
       .toArray();
   }
+
+  async getTrendingGames(timePeriodInMs, returnAmount, minimumPlayers) {
+    await this.#dbClient.get("games").aggregate([
+      this.#getGamesWithMinimumXPlayers(minimumPlayers),
+      this.#addAveragePlayersTodayProperty(),
+      this.#addAveragePlayersAtCustomDayProperty(timePeriodInMs),
```
Comment (stas-at-ibm) [inline src/infrastructure/repositories/games.repository.js:142]: `addAveragePlayersAtCustomDayProperty > addAveragePlayersInTimePeriodProperty`

File: src/infrastructure/repositories/games.repository.js [hunk 2]
```diff
@@ -134,4 +134,176 @@ export class GamesRepository {
...
+      this.#getGamesWithAveragePlayersValues(),
+      this.#addPercentagePlayerIncreaseProperty(),
+      { $unset: "averagePlayersToday" },
+      { $unset: "averagePlayersAtCustomDay" },
```
Comment (stas-at-ibm) [inline src/infrastructure/repositories/games.repository.js:146]: rename like above

File: src/infrastructure/repositories/games.repository.js [hunk 3]
```diff
@@ -134,4 +134,176 @@ export class GamesRepository {
...
+    return {
+      $match: {
+        $expr: {
+          $gt: [{ $arrayElemAt: ["$playerHistory.averagePlayers", -1] }, minimumPlayers],
+        },
+      },
+    };
+  };
+
+  #addAveragePlayersTodayProperty = () => {
```
Comment (stas-at-ibm) [inline src/infrastructure/repositories/games.repository.js:167]: ok I get everything except those two big functions :), addAveragePlayersTodayProperty, addAveragePlayersAtCustomDayProperty...to many syntax which I don't know, but I can ask gpt to explain actually

**Review by stas-at-ibm — CHANGES_REQUESTED** [review]
Great job as always. Minor things only.

File: src/utils/time.utils.spec.js [hunk 4]
```diff
@@ -86,6 +87,30 @@ describe("time.utils.js", () => {
     });
   });
 
+  describe(".daysToMs", () => {
+    let result;
+
+    describe("passing a value of 12 into the function", () => {
```
[the remark this answers is not in the document]
Comment (lukatarman) [reply src/utils/time.utils.spec.js:93]: Yeah I've been trying to do this more lately. Let me know how it is in the other PR

File: src/utils/time.utils.spec.js [hunk 5]
```diff
@@ -86,6 +87,30 @@ describe("time.utils.js", () => {
     });
   });
 
+  describe(".daysToMs", () => {
```
[the remark this answers is not in the document]
Comment (lukatarman) [reply src/utils/time.utils.spec.js:90]: :)

File: src/features/game-queries/game.queries.controller.js [hunk 6]
```diff
@@ -21,7 +21,7 @@ export class GameQueriesController {
     return await this.#gamesRepository.getGamesBySearchTerm(term);
   }
 
-  async getTrendingGames(timePeriodInMs, returnAmount = 10, minimumPlayers = 100) {
+  async getTrendingGames(timePeriodInMs, returnAmount, minimumPlayers) {
```
Comment (stas-at-ibm) [inline src/features/game-queries/game.queries.controller.js:24]: remove here also, only leave in repo othewise you pass in undefined

**Review by stas-at-ibm — APPROVED** [review]
Just one remark. You can merge after.

**Pull request merged**